nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/ConsumeJMS.java

```diff
@@ -116,18 +154,26 @@ protected JMSConsumer finishBuildingTargetResource(JmsTemplate jmsTemplate) {
     }
 
     /**
+     *
+     */
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        List<PropertyDescriptor> pd = new ArrayList<>(super.getSupportedPropertyDescriptors());
+        pd.add(ACKNOWLEDGEMENT_MODE);
+        return pd;
```

Comment:
Not an incredibly big deal, but getSupportedPropertyDescriptors() is called very often (most often in validation during a UI Refresh) which causes lots of ArrayLists to be created.  I prefer the way DistributeLoad handles this, by creating one ArrayList and reusing it, while also using an AtomicBoolean to know when it should be recreated.